cartography/intel/aws/cloudtrail_management_events.py

```diff
@@ -0,0 +1,553 @@
+            f"Failed to retrieve CloudTrail management events for region '{region}': {str(e)}"
+        )
+
+    return events
+
+
+@timeit
+def transform_cloudtrail_events_to_role_assumptions(
+    events: List[Dict[str, Any]],
+    region: str,
+    current_aws_account_id: str,
+) -> List[Dict[str, Any]]:
+    """
+    Transform raw CloudTrail events into role assumption relationships.
+
+    :type events: List[Dict[str, Any]]
+    :param events: List of raw CloudTrail events from lookup_events API
+    :type region: str
+    :param region: The AWS region where events were retrieved from
+    :type current_aws_account_id: str
+    :param current_aws_account_id: The AWS account ID being synced
+    :rtype: List[Dict[str, Any]]
+    :return: List of role assumption records in standardized format
+    """
+    role_assumptions = []
+
+    logger.info(
+        f"Transforming {len(events)} CloudTrail events to role assumptions for region '{region}'"
+    )
+
+    for event in events:
+        try:
+            assumption = _extract_role_assumption_from_event(
+                event, region, current_aws_account_id
+            )
+            if assumption:
+                role_assumptions.append(assumption)
+        except Exception as e:
+            logger.warning(
+                f"Failed to transform CloudTrail event {event.get('EventId', 'unknown')}: {str(e)}"
+            )
+            continue
+
+    logger.info(
+        f"Successfully transformed {len(role_assumptions)} role assumptions from {len(events)} events"
+    )
+    return role_assumptions
+
+
+@timeit
+def load_role_assumptions(
+    neo4j_session: neo4j.Session,
+    role_assumptions: List[Dict[str, Any]],
+    region: str,
+    current_aws_account_id: str,
+    aws_update_tag: int,
+) -> None:
+    """
+    Load role assumption relationships into Neo4j using MatchLink pattern.
+
+    Creates direct ASSUMED_ROLE relationships with aggregated properties:
+    (AWSUser|AWSRole|AWSPrincipal)-[:ASSUMED_ROLE {lastused, times_used, first_seen, last_seen}]->(AWSRole)
+
+    Assumes that both source principals and destination roles already exist in the graph.
+
+    :type neo4j_session: neo4j.Session
+    :param neo4j_session: The Neo4j session to use for database operations
+    :type role_assumptions: List[Dict[str, Any]]
+    :param role_assumptions: List of role assumption records from transform function
+    :type region: str
+    :param region: The AWS region where events were retrieved from
+    :type current_aws_account_id: str
+    :param current_aws_account_id: The AWS account ID being synced
+    :type aws_update_tag: int
+    :param aws_update_tag: Timestamp tag for tracking data freshness
+    :rtype: None
+    """
+    if not role_assumptions:
+        logger.info("No role assumption events to load")
+        return
+
+    # Aggregate role assumptions by (source, destination) pairs
+    aggregated_assumptions = _aggregate_role_assumptions(role_assumptions)
```

Comment:
This should be in the transform step. the `load()` step should not do transforms